xml/telnet_banners.xml

```diff
@@ -0,0 +1,648 @@
+    <param pos="0" name="os.product" value="ScreenOS"/>
+    <param pos="0" name="os.cpe23" value="cpe:/o:juniper:screenos:-"/>
+    <param pos="0" name="hw.vendor" value="Juniper"/>
+    <param pos="0" name="hw.device" value="Firewall"/>
+    <param pos="0" name="hw.product" value="NetScreen"/>
+  </fingerprint>
+  <fingerprint pattern="^(?:\r|\n)*KWS-1043N login:\s*$">
+    <description>Clipcomm KWS router</description>
+    <example>KWS-1043N login:</example>
+    <param pos="0" name="service.vendor" value="Clipcomm"/>
+    <param pos="0" name="service.product" value="telnet"/>
+    <param pos="0" name="hw.vendor" value="Clipcomm"/>
+    <param pos="0" name="hw.device" value="Router"/>
+    <param pos="0" name="hw.product" value="KWS-1043N"/>
+  </fingerprint>
+  <fingerprint pattern="^(?:\r|\n)*(SMCD3\w+-\w\w\w) login:\s*$">
+    <description>SMC Cable Modem</description>
+    <example>SMCD3GN2-BIZ login:</example>
+    <param pos="0" name="service.vendor" value="SMC Networks"/>
+    <param pos="0" name="service.product" value="telnet"/>
+    <param pos="0" name="hw.vendor" value="SMC Networks"/>
+    <param pos="0" name="hw.device" value="Cable Modem"/>
+    <param pos="1" name="hw.product"/>
+  </fingerprint>
+  <fingerprint pattern="^Welcome to Microsoft Telnet Service \r\n\n\rlogin:\s*$">
+    <description>Microsoft Windows</description>
+    <!-- Welcome to Microsoft Telnet Service \r\n\n\rlogin: -->
+    <example _encoding="base64">V2VsY29tZSB0byBNaWNyb3NvZnQgVGVsbmV0IFNlcnZpY2UgDQoKDWxvZ2luOgo=</example>
+    <param pos="0" name="os.vendor" value="Microsoft"/>
+    <param pos="0" name="os.family" value="Windows"/>
+    <param pos="0" name="os.product" value="Windows CE"/>
+    <param pos="0" name="os.cpe23" value="cpe:/o:microsoft:windows_ce:-"/>
+  </fingerprint>
+  <!-- The following fingerprints are for generic Broadcom hardware where the
+       vendor has left the default banner in place. These could be rebadged by
+       ZTE, CenturyLink, Sky, Huawei, etc.
+  -->
+  <fingerprint pattern="^(BCM\d+) (?:Broadband|ADSL|xDSL|DSL) Router\r\nLogin:\s*">
+    <description>OEM'd Broadcom Router</description>
+    <!-- BCM963268 Broadband Router\r\nLogin: -->
+    <example _encoding="base64" hw.product="BCM963268">QkNNOTYzMjY4IEJyb2FkYmFuZCBSb3V0ZXINCkxvZ2luOgo=</example>
+    <param pos="0" name="hw.device" value="Router"/>
+    <param pos="1" name="hw.product"/>
+  </fingerprint>
+  <fingerprint pattern="^(BCM\d+) Broadband Router\r\nTelnet is Disabled in WAN Side$">
+    <description>OEM'd Broadcom Router</description>
+    <!-- BCM963268 Broadband Router\r\nTelnet is Disabled in WAN Side -->
+    <example _encoding="base64" hw.product="BCM963268">QkNNOTYzMjY4IEJyb2FkYmFuZCBSb3V0ZXINClRlbG5ldCBpcyBEaXNhYmxlZCBpbiBXQU4gU2lkZQo=</example>
+    <param pos="0" name="hw.device" value="Router"/>
+    <param pos="1" name="hw.product"/>
+  </fingerprint>
+  <fingerprint pattern="^(?:\r|\n)*login:\s*$" flags="REG_ICASE">
```

Comment:
IIRC we stopped using `flags` where possible, because these constants are language specific and complicate integrations just using recog content.  Instead we've been using case-insensitive regular expression modifiers (`(?i:blah)`)

Reply:
Ah, thanks for reminding me!